csrc/fusion.cpp

```diff
@@ -546,12 +541,9 @@ void Fusion::registerExpr(Expr* expr) {
   for (Val* input : expr->inputs()) {
     has_tv = has_tv || input->isA<TensorView>();
     assertInContainer(input, "Input to expr is invalid, ");
-    auto uses_copy = input->uses();
-    if (std::find(uses_copy.begin(), uses_copy.end(), expr) ==
-        uses_copy.end()) {
-      uses_copy.push_back(expr);
-      input->setUses(uses_copy);
-    }
+    // Note that we don't need to add a use here for TensorView inputs since
```

Comment:
Yes, but we need to do this for scalars. `resetTvUses` doesn't grab scalar uses appearing in, e.g., `split`.

Reply:
Ah of course. Just pushed a change to addUse if it's not a TensorView.